### ISSUE TITLE
Alt-selection to disable all turrets, except selected ones

### ISSUE THREAD

**Issue by cloudziq** [ickputzdirwech/Shortcuts-ick#9]
hello,  I love this mod and I can't imagine playing without it now
There is a shortcut for disabling Artilleries, but another shortcut-type might be really helpfull, and even more usefull:
Imagine situation where all artilleries are loaded with  standard ammo type, but player goes to one of outposts, load 'atomic shell' into artillery and want ONLY this specific artillery to fire - it's very problematic when there multiple artillery ranges overlapping.  In this situation using current shortcut to disable artilleries is a bit overkill - need to disable all other turrets in range, except this one specific turret, them manually again turn them ON one by one

Normally you'll need to click many times using Artillery-remote-target, until artillery loaded with atomic-shell will fire at target

so the idea is:
  - make another selection tool which allows to choose any artillery
  - when selected, disable all other artilleries
  - upon certain time, like 5 minutes - all disabled artilleries turn ON automatically, or maybe when shortcut is used again (?)
  
  It's possible for you to make this true?  Can be made of course very similiar to this idea, hope I not asking for too much..
  Cheers, have a nice day, and thank you for keeping this mod alive ^.^

**Comment by ickputzdirwech** [comment]
Thanks for your suggestion! You want to be able to toggle artilleries based on their inventory content right? Not a type of artillery like artillery turrets or artillery wagons? Either way I don't think I will add a second shortcut. Main reason being that it is incredibly expensive to search a whole surface for all artillery.

What might be worth investigating, however, would be introducing a per player runtime setting that would allow you to select only artillery of a certain type or even with a certain inventory content. I'm getting way ahead of myself here but it might even be possible to introduce a only/except functionality with left/right click selection. What are your thoughts on this?

PS: Always happy to hear people enjoy using this mod and even some of the imo more obscure features :)

**Comment by cloudziq** [comment]
I made some mistake earlier,  I thought Artilleries MUST be toogled one by one, while it's possible to mark entire map and toggle them all at once, then turn ON one which I need to fire.   but with big selection (big rectangle) using artillery_toggle tool, it's a bit slow

Maybe it's possible to use right-mouse-button with current Artillery-Toggle tool to mark single Artillery, which become untouched - while all other artilleries will be disabled?       I don't know how it's working, you wrote its quite expensive so..
I thought iterating over Artilleries can be very fast - as there not many of them ;p       I just don't have enough knowledge about scripting for this game

In conclusion:
- maybe current tool can works a bit faster, or intruduce alt-clicking to disable all turrets, except selected one

I not sure if ammo-filter could be much usefull - making feature for one or two persons is just wasted effort?

For now, I can live with current tool, disabling all turrets, then reenable specific one I need
I just wish it can work faster on really huge maps, but thats probably not possible :)

**Comment by ickputzdirwech** [comment]
Glad you found a way to improve your workflow :)

> intruduce alt-clicking to disable all turrets, except selected one

This sounds like a reasonable request. I added this idea to my list of things I might think about in the future.

Making it any faster is not possible. I already configured it to only look for artilleries but if the area get's really big it's still a huge task for the game. Try selecting the whole world with a deconstruction planner, the game will struggle as well. Searching for all artilleries with a script might even be faster in this case, because the game only has to do it once and not every time the selected area changes. However, most of the time people don't want to toggle all artillery at the same time I guess.

**Comment by cloudziq** [comment]
yeah ability to select just one artillery, while disabling all other ones would be amazing if this is possible to do ^.^
I hope it's not too hard

`However, most of the time people don't want to toggle all artillery at the same time I guess.`
Thats why, if possible, this should be optional, or using right-mouse-button or .. whatever

Such feature can give the players control over which artillery will fire (like THIS one specific artillery just loaded with atomic or antimatter bullets ;p)
I looking forward to changes, thanks in advance. Cheers :)

**Comment by cloudziq** [comment]
Hmmm i was away from Factorio longer time, Hope this mod is not abandoned ... To bujaj się. Twój mod już nie jest potrzebny nikomu. TS
